src/main/scala/io/iohk/ethereum/consensus/ConsensusConfig.scala

```diff
@@ -0,0 +1,104 @@
+package io.iohk.ethereum.consensus
+
+import akka.util.ByteString
+import com.typesafe.config.{Config ⇒ TypesafeConfig}
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.nodebuilder.ShutdownHookBuilder
+import io.iohk.ethereum.utils.Logger
+import io.iohk.ethereum.validators.BlockHeaderValidatorImpl
+
+import scala.concurrent.duration.{FiniteDuration, _}
+
+/**
+ * Provides generic consensus configuration. Each consensus protocol implementation
+ * will use its own specific configuration as well.
+ *
+ * @param protocol Designates the consensus protocol.
+ * @param activeTimeout
+ * @param miningEnabled Provides support for generalized "mining". The exact semantics are up to the
+ *                      specific consensus protocol implementation.
+ */
+final case class ConsensusConfig(
```

Comment:
No change required, just a question: at some point we considered adding `final` to all our case classes, but decided it's not worth the hassle. Is there any other argument for this than `one shouldn't extend case classes`? What's your general position on this?